charmcraft.yaml

```diff
@@ -14,6 +14,13 @@ bases:
         channel: "22.04"
 parts:
   charm:
+    override-pull: |
+      craftctl default
+      if [[ ! -f requirements.txt ]]
+      then
+          echo 'ERROR: Use "tox run -e build" instead of calling "charmcraft pack" directly'
+          exit 1
+      fi
```

Comment:
Just to follow Shell best practices:
1) print errors to STDERR
2) nit: consider to use [Google Shell code style](https://google.github.io/styleguide/shellguide.html#environment)

```
      craftctl default
      if [[ ! -f requirements.txt ]]; then
          echo 'ERROR: Use "tox run -e build" instead of calling "charmcraft pack" directly' >&2
          exit 1
      fi
```